### ISSUE TITLE
Endless loop upon page load

### ISSUE THREAD

**Issue by uncaught** [logux/logux#14]
I'm hitting an endless state-replace-loop on page load:

![logux_endless_loop](https://user-images.githubusercontent.com/10322643/71558664-f3daa200-2a55-11ea-8d34-708960a3c297.PNG)

I tried debugging, but I can't make sense of it. It keeps bouncing into `replaceState` with an `undefined` `state`, 471 `actions` and an `undefined `pushHistory`. The last action is `@@redux/INIT`:

![logux_endless_loop2](https://user-images.githubusercontent.com/10322643/71558903-55037500-2a58-11ea-8bec-2c2b0d0f139b.PNG)

I've cleared my complete local storage and could reproduce the problem again. At first it's just a few loops, but with more actions added, it seems the list grows exponentially.

I'm on `@logux/core@0.3.5`

**Comment by oliverreich** [comment]
hey there!

first of all, I think logux is very very interesting and really well implemented. Good job! :)

For me, the problem occurs with vuex as well, although it is not necessarily an infinite loop (or better: it may depend on number of actions stored and result in call stack size problems (?)). It seems that store commit in vuex is called exponentially more often than there are actions in store (which should not happen if every action is commited only once?)

I started to chase it down by setting a breakpoint to the get() function of indexedStore (since the problem is only obvious when using indexedStore, however I assume the problem is there as well when using the in memory store but not as noticeable)

Here is what I found and where I think the problem lies:

1) `createStore$` calls `log.each()` in order to put all actions to an array `previous` (vuex.js line 475) ( `log.each()` iterates through all the actions in store every time it is called which is important later on)
2) For every action in `previous`, `process()` is called
3) `process()` calls `replay()`
4) `replay()` calls `log.each()` which in turn now calls `get()` on indexedStore and iterates through ALL actions and calls the supplied callback

As a result, `replaceState()` (and the store mutation) are called [number of actions]*[number of actions] times (or even more often). Is this expected behavior?

Interestingly this does not necessarily lead to wrong results in app state, at least not with less than a couple hundred actions. I assume this is because we replay the whole history just n times and start every iteration cycle with a fresh origin state. However, having 1000+ increment/decrement mutations on a single state property, things start to get weird and performance drops quite heavy :(

Hope my thoughts are in any way helpful. If not, I'd be happy for hints on how I can help :)
